### PR TITLE
feat(kafkatopic): creation simplification

### PR DESCRIPTION
**Pull request by ivan-savciuc**
## About this change—what it does

we already have retry logic in the client no need for the retry logic in the TF and fixes acceptance tests

### CODE REVIEW

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go [hunk 1]
```diff
@@ -289,26 +290,23 @@ func resourceKafkaTopicCreate(ctx context.Context, d *schema.ResourceData, m int
...
-		ServiceName:   serviceName,
-		CreateRequest: createRequest,
+	err = m.(*aiven.Client).KafkaTopics.Create(
+		project,
+		serviceName,
+		createRequest,
+	)
+	if err != nil && !aiven.IsAlreadyExists(err) {
+		diag.FromErr(err)
 	}
```
Comment (byashimov) [inline internal/sdkprovider/service/kafkatopic/kafka_topic.go:300]: Must return after error.

**Review by Serpentiel — DISMISSED** [review]
lgtm, @byashimov wdyt?

**Comment by ivan-savciuc** [comment]
Tests are still failing, lets catch up to discuss this issue.

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go [hunk 2]
```diff
@@ -289,26 +290,23 @@ func resourceKafkaTopicCreate(ctx context.Context, d *schema.ResourceData, m int
...
-
-	// nolint:staticcheck // TODO: Migrate to helper/retry package to avoid deprecated WaitForStateContext.
-	_, err = w.Conf(timeout).WaitForStateContext(ctx)
-	if err != nil {
-		return diag.FromErr(err)
+	// Invalidates cache for the topic
+	// We need this only for acceptance tests
+	if os.Getenv("TF_ACC") == "1" {
+		DeleteTopicFromCache(project, serviceName, topicName)
 	}
```
Comment (byashimov) [inline internal/sdkprovider/service/kafkatopic/kafka_topic.go:306]: Please don't mix tests with the business code. Delete topic from cache in test.

**Review by byashimov — CHANGES_REQUESTED** [review]

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go (hunk 2 above)
Comment (byashimov) [inline internal/sdkprovider/service/kafkatopic/kafka_topic.go:306]: This code must be a rudimentary. [This](https://github.com/aiven/terraform-provider-aiven/blob/7b54651089fce0a9cc45db4a36c68291f521a407/internal/sdkprovider/service/kafkatopic/kafka_topic_test.go#L392-L392) should be enough.

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go (hunk 1 above)
Comment (ivan-savciuc) [reply internal/sdkprovider/service/kafkatopic/kafka_topic.go:300]: good point

File: internal/sdkprovider/service/kafkatopic/kafka_topic.go (hunk 2 above)
Comment (ivan-savciuc) [reply internal/sdkprovider/service/kafkatopic/kafka_topic.go:306]: ok, I'm going to try this one

**Review by byashimov — APPROVED** [review]
lgtm

**Pull request merged**